src/settings.ts

```diff
@@ -56,5 +56,15 @@ export default class CopyPublishUrlSettingTab extends PluginSettingTab {
                         await this.plugin.saveSettings();
                     });
             });
+        new Setting(containerEl)
+            .setName('Enable in Context Menu')
+            .setDesc('Allow copy publish in context menu.')
+            .addToggle(toggle => {
+                toggle.setValue(this.plugin.settings.enableContext)
+                toggle.onChange(value =>{
```

Comment:
Please make this async. `saveSettings()` calls `saveData()`, [which is async](https://github.com/obsidianmd/obsidian-api/blob/5687a75ecfdb78b6a832127cf6f9229e27e196a0/obsidian.d.ts#L2345).